### ISSUE TITLE
Towny and Modern LWC Issue

### ISSUE THREAD

**Issue by Lunna123** [Tsuser1/Modern-LWC#98]
**LWC Version (/lwc admin version):** v 2.1.2
**Server version (/version):** git-Paper-610

**Short five word summary of issue:** 
Outsiders can /cprivate unlocked chests

**Detailed description of problem:**
I'm not sure if anyone else has encountered this issue with Towny and Modern LWC, but if a Town has an unlocked chest placed on a town claim, anyone can come along and lock that chest using /cprivate, even if they aren't part of that Town.

Ultimately what happens is neither Town members nor the owner of the protected chest can open the chest (if outsiders can't open Town chests). 

If Town members try to open the chest, it says "This chest is locked with a magical spell", and if the chest owner tries to open the chest it says "[Towny] Outsiders cannot switch here" or something like that.

Ideally, I think, if you aren't part of said town, you shouldn't be able to protect unlocked chests on its claims. 

**If there was an error in console, put it here:**
No errors.

**Comment by pop4959** [comment]
It sounds like it is unintended, so it would make sense to prevent outsiders from establishing locks, as you mentioned. I would be willing to look into it when I have time.

It would help me if you can provide me a link to the version of Towny you are trying to use, and give step by step instructions on how to make a town such as the one that you are having issues with. I have not had any experience working with the LWC-Towny integration yet, so please bear with me. :)

**Comment by NyannyWu** [comment]
I can help with a 1.13 towny build
https://cdn.discordapp.com/attachments/487025202539266048/562701987322396702/Towny.jar

Steps to create a town
/t new name
/t deposit 100000 (to put money into town bank)
/t claim (to claim new plots)
/t invite name (to invite someone to a town)
/plot fs 0 (to sell a plot for 0)
/plot claim to claim a plot

Also, each town sells plots to anyone, even if user doesnt belong to a towny buying an embassy plot (/plot set embassy), I think the check should be

If plot is owned by someone, only him and friends a can lock stuff
if plot is unowned but is inside a town, only members with certain ranks, like mayor/assistant should be able to lock.

But if it's complicated just to members of a town should work, i think...

**Comment by pop4959** [comment]
@NyannyWu Thanks for the information. Also, can you link to the original location where you downloaded the plugin?

**Comment by NyannyWu** [comment]
Towny Advanced official discord: https://discord.gg/U6FA8CG 

You can find this link  on their website: http://towny.palmergames.com/

**Comment by pop4959** [comment]
@NyannyWu Thanks. I've seen that towny site before but I wasn't sure if that was the official source. Discord should help as well!

**Comment by Tombstone1973** [comment]
I don't know if this already exists, and was told to add it here by Tsuser1. The ability to have chest protections automatically clean themselves up if a player hasn't logged in for x amount of days (configurable in the config) and to automatically remove chest protections in town claimed areas if that player is no longer in the town?

**Comment by pop4959** [comment]
@Tombstone1973 Please don't hijack this issue, make a new one if you want your idea discussed. Thanks

**Comment by Tombstone1973** [comment]
@pop4959 I was told to post it here by tsuser over discord.

**Comment by pop4959** [comment]
It sounds like a different suggestion, so it should be in its own issue

**Comment by Tombstone1973** [comment]
Well, one thing would fall under towny integration. The other not so much. Either way, don't shoot the dude listening to directions.

**Comment by pop4959** [comment]
I'm not trying to argue with you over the confusion, but this issue is relevant only to what was originally posted. It will be closed when the original issue has been looked at. It's in your best interest to make a new issue if you want to have proper discussion of it. That's just how it is, no need to get offended. Thank you

**Comment by Lunna123** [comment]
Sorry for the lack of replies. Not sure if this has been solved yet? 

I'm impatient and ended up fixing this myself. :p Only caveat to my solution is that it uses the same error message as the message for if townyBorders in core.yml is set to true (i.e. cannot claim outside towny borders). I don't use this setting so I just changed the message in the locale to suit this error. Also haven't done any testing beyond the conditions I use for my server.

**Comment by chbohn** [comment]
@Lunna123 Sorry, I've not had time to address the issue myself yet, due to a busy schedule. Would you be able to make a pull request containing your fix, if you don't mind? I can review and test it to make sure things are all right, and adjust the config/messages as required. Thanks!

**Comment by pop4959** [comment]
@Lunna123 Sorry, I've not had time to address the issue myself yet, due to a busy schedule. Would you be able to make a pull request containing your fix, if you don't mind? I can review and test it to make sure things are all right, and adjust the config/messages as required. Thanks!

**Comment by pop4959** [comment]
@Lunna123 I have implemented a fix for this now (in the latest dev build). I have only done trivial testing (try to lock as member and non-member) though, so if you feel you have a more thorough fix, you are still free to make a pull request. I'll close this issue for now though since the problem at hand has been addressed.

Any other problems/suggestions should be posted in a new issue. I am closing this now.
